src/runtime/hexagon_remote/host_malloc.cpp

```diff
@@ -43,7 +46,28 @@ struct ion_handle_data {
 #define ION_IOC_FREE _IOWR('I', 1, ion_handle_data)
 #define ION_IOC_MAP _IOWR('I', 2, ion_fd_data)
 
-ion_user_handle_t ion_alloc(int ion_fd, size_t len, size_t align, unsigned int heap_id_mask, unsigned int flags) {
```

Comment:
I think that all of the old changes to ion_* should be reverted, and keep all of the branching to the new API in halide_hexagon_host_*. Right now, it's kind of done in both places, but not completely in either place, which is confusing/complicated.

Reply:
in progress, thx for suggestion